src/reporters/remote_reporter.js

```diff
@@ -65,6 +65,10 @@ export default class RemoteReporter {
         } else {
             this._metrics.reporterSuccess.increment(response.numSpans);
         }
+
+        if (callback) {
+            callback();
+        }
```

Comment:
Also, before calling sender.close() we need to call sender.flush(). And then I would reconsider having the callback in sender.close(), because it becomes synch method.